openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -124,9 +156,33 @@ class AccessTokenExchangeView(_DispatchingView):
     """
     dot_view = auth_exchange_views.DOTAccessTokenExchangeView
 
+    def dispatch(self, request, *args, **kwargs):
```

Comment:
Looks OK from a security perspective. I don't understand the full context, but I think that the ability to ask for and receive a JWT instead of an opaque bearer token (in whatever flow) seems fine in general.